src/test/java/com/karankumar/bookproject/backend/entity/UserTest.java

```diff
@@ -115,4 +122,16 @@ void testPasswordWithNoSpecialCharacterIsInvalid() {
         // then
         Assertions.assertEquals(initialNumberOfUsers, userRepository.count());
     }
+
+    @Test
+    void testUserWithInvalidMail() {
```

Comment:
Could you rename this method to be more explicit about what is being tested? While we know a user with an invalid email address is being tested, we do not know why exactly the email is invalid? 

This is important because other invalid email tests are required.

Reply:
Yes, I fixed and added other tests.